File: clients/javascript/src/models/msgTopicConfigureIn.ts
// this file is @generated

export interface MsgTopicConfigureIn {
    partitions: number;
}

export interface MsgTopicConfigureIn_ {
    topic: string;
    partitions: number;
}

export const MsgTopicConfigureInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgTopicConfigureIn_ {
        return {
            topic: object['topic'],
            partitions: object['partitions'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgTopicConfigureIn_): any {
        return {
            'topic': self.topic,
            'partitions': self.partitions,
        };
    }
}

File: clients/rust/src/models/msg_topic_configure_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize)]
pub struct MsgTopicConfigureIn {}

impl MsgTopicConfigureIn {
    pub fn new() -> Self {
        Self {}
    }
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub(crate) struct MsgTopicConfigureIn_ {
    pub topic: String,

    pub partitions: u16,
}
